hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java

```diff
@@ -162,6 +162,10 @@ private RecordStore getRecordStoreOrNull() {
     }
 
     protected boolean isNativeInMemoryFormat() {
+        if (mapContainer == null) {
```

Comment:
Can this be related to the fixes we did for the `MapProxy` and `TransactionalMapProxy` to not create a `MapContainer`? I guess it happens on members, which hold no data for a map?

Reply:
I'm not sure I'm following this - the `com.hazelcast.map.impl.operation.MapOperation#innerBeforeRun` should create the map container even if it does not exist yet. Or am I wrong?

Reply:
I didn't have a look at the context. I tried to describe the general scenario where a `MapContainer` can be `null` on a subset of members.

Reply:
@mmedenjak This happens if there is no quorum, in this case the beforeRun is not executed.

https://github.com/tombujok/hazelcast/blob/16bd03c7794322b5989d1bd5641c80352c1117cb/hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java#L186-L188

Any other idea how to fix it? I kind of dislike this returning `false`

Reply:
Umph, that's a tough one :) Here's one idea (probably not that good):
```
    @Override
    public void onExecutionFailure(Throwable e) {
        if (!(e instanceof QuorumException) && isNativeInMemoryFormat()) {
            // This is required since if the returnsResponse() method returns false there won't be any response sent
            // to the invoking party - this means that the operation won't be retried if the exception is instanceof
            // HazelcastRetryableException
            sendResponse(e);
        } else {
            super.onExecutionFailure(e);
        }
    }
```
Or just add some comment so we know what it's about and don't remove that `if`.